scipy/signal/tests/test_signaltools.py

```diff
@@ -267,6 +267,19 @@ def test_fillvalue(self):
                    [32, 46, 67, 62, 48]])
         assert_array_equal(c, d)
 
+    def test_fillvalue_direct_cast(self):
+        # These is an example that used to work, but it seems odd enough
+        # to disallow directly?
```

Comment:
OK, I guess the question is how much is OK to break odd cases here without deprecation. For example I could do:
```
# Preserve all but broken types that numpy cannot cast to a native dtype:
np.array(fillvalue).astype(actual_type)
# Additionally, do not preserve some stranger casts (complex fill_value -> non-complex):
np.array(fillvalue, dtype=actual_type)
```
Also one could:
  1. Add a deprecation and fall back to old behaviour
  2. Add a deprecation to enforce that fillvalue is a scalar.